### DISCUSSION TITLE
What does emacs-plus have that emacs doesn't?

### DISCUSSION THREAD

**Discussion by renatoathaydes** [d12frosted/homebrew-emacs-plus#385] [Q&A] [answered]
Hi!

I decided to install emacs-plus@28 to see if it fixes the horrible flickering vanilla emacs has on Mac (both my machines have the exact same problem).

It looks great, but I don't see anything extra compared to emacs (which is good!). Just wanted to know if that's correct, or if there's some packages that are installed extra?

Thanks!

**Comment by mkvoya** [comment, marked as answer]
In my understanding, emacs-plus is simply the vanilla emacs with some small patches (and icons!). You can check the directory https://github.com/d12frosted/homebrew-emacs-plus/tree/master/patches/emacs-28, which contains the available patches on emacs-28. All of them are exposed by the options in the README file.

**Reply by d12frosted** [reply]
Yes, you are right 😸 It's just a homebrew formula for GNU Emacs with some minor patches (some of them are optional) and several optional icons. How does it differ from 'official' homebrew `emacs` formula? `emacs-plus` provides both bin and app and have options not available in `emacs` formula (see readme for the full list of options).

How does it differ from Emacs for Mac `emacs-mac`? The latter is a fork of GNU Emacs that aim to better integrate Emacs into macos ecosystem. It has its own set of pros and cons. You can read a little bit more here - https://github.com/d12frosted/homebrew-emacs-plus/issues/1.

**Reply by renatoathaydes** [reply]
Ok, great to know! It looks good and that's exactly what I was looking for, though the toolbar moving into the title bar looks a bit weird (not a problem as I normally use emacs fullscreen so I don't normally see that).

BTW it does fix the MacOS flickering (more like a full screen "refresh") that makes emacs 27.x totally unusable on Mac.

**Reply by d12frosted** [reply]
> It looks good and that's exactly what I was looking for

Great! Glad to hear.

> BTW it does fix the MacOS flickering (more like a full screen "refresh") that makes emacs 27.x totally unusable on Mac.

Ah, this was so annoying... it took quite a while to fix that, but folks working on GNU Emacs did a great job and now we can enjoy Emacs 😄

**Reply by renatoathaydes** [reply]
I went through the "News" on the 28 release, it's insane how much stuff they've done (the GNU Emacs maintainers)! Seriously good work.
